components/places/src/api/matcher.rs

```diff
@@ -311,95 +309,95 @@ impl<'query, 'conn> OriginOrUrl<'query, 'conn> {
+        WHERE host BETWEEN :searchString AND :searchString || X'FFFF'
+        GROUP BY host
+        HAVING host_frecency >= :frecencyThreshold
+        UNION ALL
+        SELECT host,
+                TOTAL(frecency) AS host_frecency,
+                (SELECT TOTAL(foreign_count) > 0 FROM moz_places
+                WHERE moz_places.origin_id = moz_origins.id) AS bookmarked
+        FROM moz_origins
+        WHERE host BETWEEN 'www.' || :searchString AND 'www.' || :searchString || X'FFFF'
+        GROUP BY host
+        HAVING host_frecency >= :frecencyThreshold
+    ) AS grouped_hosts
+    JOIN moz_origins ON moz_origins.host = grouped_hosts.host
+    ORDER BY frecency DESC, id DESC
+    LIMIT 1
+";
+
 impl<'query, 'conn> Matcher for OriginOrUrl<'query, 'conn> {
     fn search(&self, _: u32) -> Result<Vec<SearchResult>> {
-        let mut results = Vec::new();
-        if looks_like_origin(self.query) {
-            let mut stmt = self.conn.db.prepare(
-                "
-                SELECT IFNULL(:prefix, prefix) || moz_origins.host || '/' AS url,
-                       moz_origins.host || '/' AS displayURL,
-                       frecency,
-                       bookmarked,
-                       id,
-                       :searchString AS searchString
-                FROM (
-                  SELECT host,
-                         TOTAL(frecency) AS host_frecency,
-                         (SELECT TOTAL(foreign_count) > 0 FROM moz_places
-                          WHERE moz_places.origin_id = moz_origins.id) AS bookmarked
-                  FROM moz_origins
-                  WHERE host BETWEEN :searchString AND :searchString || X'FFFF'
-                  GROUP BY host
-                  HAVING host_frecency >= :frecencyThreshold
-                  UNION ALL
-                  SELECT host,
-                         TOTAL(frecency) AS host_frecency,
-                         (SELECT TOTAL(foreign_count) > 0 FROM moz_places
-                          WHERE moz_places.origin_id = moz_origins.id) AS bookmarked
-                  FROM moz_origins
-                  WHERE host BETWEEN 'www.' || :searchString AND 'www.' || :searchString || X'FFFF'
-                  GROUP BY host
-                  HAVING host_frecency >= :frecencyThreshold
-                ) AS grouped_hosts
-                JOIN moz_origins ON moz_origins.host = grouped_hosts.host
-                ORDER BY frecency DESC, id DESC
-                LIMIT 1
-            ",
-            )?;
-            let params: &[(&str, &dyn rusqlite::types::ToSql)] = &[
-                (":prefix", &rusqlite::types::Null),
-                (":searchString", &self.query),
-                (":frecencyThreshold", &-1i64),
-            ];
-            for result in stmt.query_and_then_named(params, SearchResult::from_origin_row)? {
-                results.push(result?);
-            }
+        Ok(if looks_like_origin(self.query) {
+            self.conn.query_rows_and_then_named(
```

Comment:
I'd have thought these would make sense with `query_rows_and_then_named_cached`?

Reply:
Oh, sure. We also should consider increasing the cache size, since it's only 16 items. In places I can't seem to find the cache size (and, tbh, it seems like it might be unbounded?).